x-pack/plugins/ingest_manager/common/types/models/epm.ts

```diff
@@ -256,10 +256,32 @@ export enum DefaultPackages {
   endpoint = 'endpoint',
 }
 
+export interface Mappings {
+  dynamic?: boolean | 'strict';
+  date_detection?: boolean;
+  dynamic_templates?: any[];
+  properties: Record<string, Mapping>;
+}
+
+interface Mapping {
+  type: string;
+  properties?: Record<string, Mapping>;
+}
+
 export interface IndexTemplate {
   order: number;
   index_patterns: string[];
   settings: any;
-  mappings: object;
+  mappings: Mappings;
   aliases: object;
 }
+
+export interface TemplateRef {
+  templateName: string;
+  indexTemplate: IndexTemplate;
+}
+
+export interface CurrentIndex {
```

Comment:
Can you move this to `x-pack/plugins/ingest_manager/server/services/epm/elasticsearch/template/template.ts`? I think that's the only place it's used and I don't initially see how `CurrentIndex` is a top-level concept in EPM.